thoth/solver/python/python.py

```diff
@@ -433,6 +434,28 @@ def resolve(requirements, *, index_urls, python_version, exclude_packages, trans
         packages.sort(key=lambda p: (p.count("."), p))
         item["packages"] = packages
 
+    run_command("touch ./temp-requirements.txt")
+
+    for dependency in result["tree"]:
```

Comment:
Thanks for digging into this. This is really low-level and hardcore part of the system.

The approach might require some more effort to have it slightly restructured. The thing is that solver is run thousands of times in the cluster and each artifact download costs us a lot of time. This change could eventually land in the code that does the actual dependency extraction to make sure the artifact is downloaded really just once and the corresponding data are extracted from it (per artifact). It might happen that two different builds can have different metadata which we should be aware of (hence this could land in a different part of the codebase). I'm open to discuss the approach on a call if you want. Thanks again for your contribution and time spent on this.